code/red-black-tree/ViewModel/src/main/java/ru/unn/agile/redblacktree/viewmodel/ViewModel.java

```diff
@@ -0,0 +1,207 @@
+        return findElementToTreeDisabled.get();
+    }
+    public BooleanProperty removeElementToTreeDisabledProperty() {
+        return removeElementToTreeDisabled;
+    }
+    public final boolean isRemoveElementToTreeDisabled() {
+        return removeElementToTreeDisabled.get();
+    }
+
+    public StringProperty resultFindProperty() {
+        return resultFind;
+    }
+    public final String getResultFind() {
+        return resultFind.get();
+    }
+    public StringProperty resultRemoveProperty() {
+        return resultRemove;
+    }
+    public final String getResultRemove() {
+        return resultRemove.get();
+    }
+
+    public ViewModel() {
+        addField.set("");
+        findInsertField.set("");
+        removeInsertField.set("");
+        resultFind.set("");
+        resultRemove.set("");
+
+        status.set(Status.WAITING.toString());
+
+        BooleanBinding couldAddElementToTree = new BooleanBinding() {
+            {
+                super.bind(addField);
+            }
+            @Override
+            protected boolean computeValue() {
+                return getInputStatus() == Status.READY;
+            }
+        };
+        addElementToTreeDisabled.bind(couldAddElementToTree.not());
+
+        BooleanBinding couldFindElementToTree = new BooleanBinding() {
+            {
+                super.bind(findInsertField);
+            }
+            @Override
+            protected boolean computeValue() {
+                return getInputStatus() == Status.READY;
+            }
+        };
+        findElementToTreeDisabled.bind(couldFindElementToTree.not());
+
+        BooleanBinding couldRemoveElementToTree = new BooleanBinding() {
+            {
+                super.bind(removeInsertField);
+            }
+            @Override
+            protected boolean computeValue() {
+                return getInputStatus() == Status.READY;
+            }
+        };
+        removeElementToTreeDisabled.bind(couldRemoveElementToTree.not());
+
+        // Add listeners to the input text fields
+        final List<StringProperty> fields = new ArrayList<StringProperty>() { {
+            add(addField);
+            add(findInsertField);
+            add(removeInsertField);
+        } };
+
+        for (StringProperty field : fields) {
+            final ValueChangeListener listener = new ValueChangeListener();
+            field.addListener(listener);
+            valueChangedListeners.add(listener);
+        }
+    }
+
+    public StringProperty addFieldProperty() {
+        return addField;
+    }
+    public StringProperty findInsertFieldProperty() {
+        return findInsertField;
+    }
+    public StringProperty removeInsertFieldProperty() {
+        return removeInsertField;
+    }
+
+    public StringProperty statusProperty() {
+        return status;
+    }
+    public final String getStatus() {
+        return status.get();
+    }
+
+    private Status getInputStatus() {
```

Comment:
Not a good idea to validate all inputs at once, but at least name it like `getInputsStatus`

Reply:
Resolved